transbot/models/res_config.py

```diff
@@ -0,0 +1,128 @@
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields, api, exceptions, _
+try:
+    from txlib.http.auth import BasicAuth
+    from txlib.http.http_requests import HttpRequest
+    from txlib.http.exceptions import AuthorizationError
+except ImportError:
+    BasicAuth = HttpRequest = AuthorizationError = None
+from requests.exceptions import ConnectionError
+try:
+    from github import Github
+except ImportError:
+    Github = None
+
+
+class TransbotConfigSettings(models.TransientModel):
+    _name = 'transbot.config.settings'
+    _inherit = 'res.config.settings'
+
+    transifex_url = fields.Char(string="Transifex base URL")
+    transifex_user = fields.Char(string="Transifex user")
+    transifex_password = fields.Char(string="Transifex password")
+    transifex_organization = fields.Char(
+        string="Transifex organization",
+        help="This is the slug of the Transifex organization to be used for "
+             "the creation of new projects, and for the checking of new "
+             "translations made")
+    transifex_team = fields.Integer(
+        string="Transifex team number",
+        help="This is the number of the team responsible for making the "
+             "transbot translations")
+    github_token = fields.Char(string="Github token")
+
+    def get_github_connection(self):
+        rec = self._get_parameter('github.token')
+        if not rec:
+            raise exceptions.Warning(_('There is no Github token defined'))
+        else:
+            return Github(login_or_token=rec.value)
+
+    def get_transifex_connection(self):
+        transifex_url = self._get_parameter('transifex.url').value
+        transifex_user = self._get_parameter('transifex.user').value
+        transifex_password = self._get_parameter('transifex.password').value
+        if not transifex_url or not transifex_user or not transifex_password:
+            raise exceptions.Warning(_('Transifex credentials are not fully '
+                                       'defined'))
+        else:
+            auth = BasicAuth(transifex_user, transifex_password)
+            return HttpRequest(transifex_url, auth=auth)
+
+    def _get_parameter(self, key):
+        param_obj = self.env['ir.config_parameter']
+        rec = param_obj.search([('key', '=', key)])
+        return rec or False
+
+    @api.multi
+    def get_default_parameters(self):
```

Comment:
I would suggest factoring the computation that is copied a few times, for example:
```python 
    @api.multi
    def get_default_parameters(self):
        def get_value(key, default=''):
            rec = self._get_parameter('transifex.url')
            return rec and rec.value or default
        return {
            'transifex_url'': get_value('transifex.url', 'www.transifex.com'),
            'transifex_user'': get_value('transifex.user'),
            'transifex_password'': get_value('transifex.password'),
            'transifex_organization'': get_value('transifex.organization'),
            'transifex_team'': get_value('transifex.team'),
            'github_token'': get_value('github.token'),
        }
```